src/fn_native.rs

```diff
@@ -141,4 +179,16 @@ impl CallableFunction {
     pub fn from_method(func: Box<FnAny>) -> Self {
         Self::Method(func.into())
     }
+
+    #[cfg(feature = "sync")]
+    /// Create a new `CallableFunction::Plugin`.
+    pub fn from_plugin(plugin: impl PluginFunction + 'static + Send + Sync) -> Self {
```

Comment:
Use `Self::Plugin(plugin.into())` to avoid multiple versions.

Reply:
Not quite:
```
182 |         Self::Plugin(plugin.into())
    |                      ^^^^^^^^^^^^^ the trait `std::convert::From<impl PluginFunction + 'static + Send + Sync>` is not implemented for `std::rc::Rc<dyn plugin::PluginFunction>`
    |
    = note: required because of the requirements on the impl of `std::convert::Into<std::rc::Rc<dyn plugin::PluginFunction>>` for `impl PluginFunction + 'static + Send + Sync`
```

Reply:
Ah!  OK!